aws-opentelemetry-distro/tests/amazon/opentelemetry/distro/patches/test_botocore_sns.py

```diff
@@ -0,0 +1,42 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+import botocore.session
+from moto import mock_aws
+
+from amazon.opentelemetry.distro.patches._instrumentation_patch import apply_instrumentation_patches
+from opentelemetry.instrumentation.botocore import BotocoreInstrumentor
+from opentelemetry.test.test_base import TestBase
+
+
+# pylint: disable=C0103
+class TestSnsExtension(TestBase):
```

Comment:
It looks to me like this logic is overlapping with a) patch unit tests and b) contract tests, but not really adding anything new to the mix, but does bring in more dependencies.

Why do we think this is needed? We don't do this for SQS (we only test in `test_instrumentation_patch.py`). I would prefer to keep tests comparable across patches, to avoid confusion.

Reply:
Will remove it as discussed offline:

> I was thinking that we will move all these patches to upstream one day. The upstream also need unit tests which has it’s own setup (for example [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/7bddbb54195be4004275c83ef2592b362e443dc8/instrumentation/opentelemetry-instrumentation-botocore/tests/test_botocore_sns.py#L33)). I want to implement these tests along with downstream contribution, so that we can simply move to upstream without rewrite all them in the future.

> When you go upstream you can implement the unit tests based on whatever is appropriate/required for upstream to accept. For our repo, we have our own testing framework.